### DISCUSSION TITLE
[Bug]: No network

### DISCUSSION THREAD

**Discussion by lemaitreh** [neurodesk/neurodesk.github.io-hugo#432] [Bug Report]
### What platform are you seeing the problem on?

Cloud

### What happened

Hello,

It seems there are some network issues on neurodesk play:

(base) jovyan@neurodesktop-:~$ ping google.com
ping: google.com: Temporary failure in name resolution
(base) jovyan@neurodesktop-:~$ ping google.com
PING google.com (142.251.36.142) 56(84) bytes of data.
From 10.42.0.1 (10.42.0.1) icmp_seq=1 Destination Port Unreachable

Best,

Hervé Lemaître

**Comment by stebo85** [comment]
Dear @lemaitreh,

What you are seeing is a security measure to keep crypto miners out of our platform. We block ICMP, so this means ping is not working. However, curl and google is working:

<img width="1850" alt="image" src="https://github.com/user-attachments/assets/0412822a-ea63-4300-a519-d4fc9d829799">


Does this solve your problem?

**Comment by lemaitreh** [comment]
![Capture d’écran 2024-08-20 111558](https://github.com/user-attachments/assets/b01d8554-214d-4bd8-942b-1cf090b59663)


It seems to be intermittent. It may be a problem with the DNS server as it stops responding from time to time (see screenshot).

It's the same behavior for US or European locations.

**Reply by stebo85** [reply]
Dear @lemaitreh ,

we tweaked the DNS server - can you test if it's working now for you? The download works for me:
<img width="853" alt="image" src="https://github.com/user-attachments/assets/29d4418b-158d-4acf-819a-d3b05781b6c6">

We still need to block ICMP, so ping will not work to keep the environment save and working for all users.

Can you let us know what your exact use-case is? It's probably not pinging google or downloading rocky linux?

Kind regards
Steffen

**Comment by lemaitreh** [comment]
I'm guessing you put a limitation in the download somewhere, as everything seems to work until I start using Datalad.

I want to use it for teaching purposes for a tutorial on how to process MRI images. I wanted to use datalad to download one dataset and to process it with fsl.

![Capture d’écran 2024-08-21 112016](https://github.com/user-attachments/assets/7028f2b8-e58e-4eab-a249-6fa1ecb286af)

**Reply by stebo85** [reply]
Cool - we could get this working and unblock it. What datalad commands are you using?

Thank the stupid crypto world for all the security measures we have to put in place to offer such a service ... and sometimes these block legitimate use cases

**Reply by stebo85** [reply]
Ok, saw the commands in the screenshot now - we have a look!

**Reply by stebo85** [reply]
@lemaitreh, fixed :)
<img width="725" alt="image" src="https://github.com/user-attachments/assets/71796908-f65c-4cdd-b4e6-ca94058f6ff6">

please let us know if you find any other legitimate use cases that we need to unblock!

**Reply by lemaitreh** [reply]
Thanks a lot. It works like a charm now ;-)

**Reply by stebo85** [reply]
@lemaitreh - great to hear it's working :) For workshops and teaching with Neurodesk we recommend to setup a dedicated instance that offers authentication, shared storage between participants, more compute and storage resources and no network filtering - let us know if this would be interesting for you!

**Reply by lemaitreh** [reply]
Hi,
I'm planning to do a basic task fmri tutorial for 15 personnes using datalad and fsl mainly on a single subject dataset. Do you think I need the dedicated instance or can they just use the standard neurodesk on their computers?
Best

**Reply by stebo85** [reply]
That could work on the play instances assuming other people are not using it for similar sized workshops at the same time. 

Otherwise users can also install Neurodesk locally on their computer and you keep the play instance only for people where you run into trouble with docker/podman installs.

Another option could be to use EGI  or commercial cloud resources - lots of possibilities :)

Let us know how you go and if you need any help
